### PR TITLE
TDR 2826 remove all english open records

### PR DESCRIPTION
**Pull request by MancunianSam**
Remove allEnglish option and initialOpenRecords parameters.

The controller tests are now done in a loop with the additional metadata set to true and false. It looks like a large change but it's mostly whitespace.

This will need https://github.com/nationalarchives/tdr-consignment-api/pull/547 to be merged and a new version of the generated graphql project before it can be merged.

### CODE REVIEW

**Review by TomJKing — COMMENTED** [review]
Looks good. Thanks. I've not approved it as there is a merge conflict. 

Also, pretty sure this change will break the e2e tests. Could you make sure there is a PR ready to update the e2e tests based on this change.

File: app/controllers/TransferAgreementComplianceController.scala [hunk 1]
```diff
@@ -21,28 +22,52 @@ class TransferAgreementComplianceController @Inject() (
...
+
+  private def form: Form[TransferAgreementComplianceData] =
+    if (blockClosureMetadata && blockDescriptiveMetadata) {
+      transferAgreementFormWithOpenRecords
+    } else {
+      transferAgreementForm
+    }
+
+  val blockClosureMetadata: Boolean = configuration.get[Boolean]("featureAccessBlock.closureMetadata")
+  val blockDescriptiveMetadata: Boolean = configuration.get[Boolean]("featureAccessBlock.descriptiveMetadata")
```
Comment (vimleshtna) [inline app/controllers/TransferAgreementComplianceController.scala:62]: It's just a suggestion. Feel free to ignore it.

Rename `FrontEndInfoConfiguration` to `ApplicationConfig` and get these config values from this class:

```

class ApplicationConfig @Inject()(configuration: Configuration) {

  private def get(location: String) = configuration.get[String](location)

  def frontEndInfo: FrontEndInfo = FrontEndInfo(
    get("consignmentapi.url"),
    get("environment"),
    get("region"),
    get("upload.url")
  )

  val blockClosureMetadata: Boolean = configuration.get[Boolean]("featureAccessBlock.closureMetadata")
  val blockDescriptiveMetadata: Boolean = configuration.get[Boolean]("featureAccessBlock.descriptiveMetadata")
}

```
Now, we can use these variables on the controllers. Actually, I don't want to hardcode the same string everywhere.

File: test/testUtils/FrontEndTestHelper.scala [hunk 2]
```diff
@@ -83,6 +85,18 @@ trait FrontEndTestHelper extends PlaySpec with MockitoSugar with Injecting with
     }
   }
 
+  def getConfig(blockAdditionalMetadata: Boolean): Configuration = {
+    val config: Map[String, ConfigValue] = ConfigFactory
+      .load()
+      .withValue("featureAccessBlock.closureMetadata", ConfigValueFactory.fromAnyRef(blockAdditionalMetadata.toString))
+      .withValue("featureAccessBlock.descriptiveMetadata", ConfigValueFactory.fromAnyRef(blockAdditionalMetadata.toString))
+      .entrySet()
```
Comment (vimleshtna) [inline test/testUtils/FrontEndTestHelper.scala:93]: I'm not sure if we already have test scenarios to test if `featureAccessBlock.closureMetadata` is true but `featureAccessBlock.descriptiveMetadata` is false or other way around. So in that case, the controllers (`TransferAgreementComplianceController` and `TransferAgreementPrivateBetaController`) should show the `transferAgreementForm`.

**Review by vimleshtna — COMMENTED** [review]
Looks good, just a few suggestions, thanks.

File: test/testUtils/FrontEndTestHelper.scala (hunk 2 above)
Comment (MancunianSam) [reply test/testUtils/FrontEndTestHelper.scala:93]: Ah good point yeah. I've added in a table with each of the options for the two booleans and run the tests for all of them. It's in a loop so when we get rid of the feature block, we can just remove the outer loop.

File: app/controllers/TransferAgreementComplianceController.scala (hunk 1 above)
Comment (MancunianSam) [reply app/controllers/TransferAgreementComplianceController.scala:62]: Yep, good point. I've moved these out the config, updated the controllers and updated the tests. I've also updated the FileCheckResults controller which is also using these feature block variables.

**Comment by MancunianSam** [comment]
I've created a draft PR for the E2E tests but that won't work until this is merged https://github.com/nationalarchives/tdr-e2e-tests/pull/421

File: app/controllers/FileChecksResultsController.scala [hunk 3]
```diff
@@ -20,16 +19,14 @@ class FileChecksResultsController @Inject() (
     val graphqlConfiguration: GraphQLConfiguration,
     val consignmentService: ConsignmentService,
     val consignmentStatusService: ConsignmentStatusService,
-    val frontEndInfoConfiguration: FrontEndInfoConfiguration,
-    val configuration: Config
+    val frontEndInfoConfiguration: ApplicationConfig,
+    val applicationConfig: ApplicationConfig
```
Comment (vimleshtna) [inline app/controllers/FileChecksResultsController.scala:23]: You can remove the old one `frontEndInfoConfiguration`.

File: app/controllers/FileChecksController.scala [hunk 4]
```diff
@@ -19,7 +19,7 @@ class FileChecksController @Inject() (
     val graphqlConfiguration: GraphQLConfiguration,
     val keycloakConfiguration: KeycloakConfiguration,
     val consignmentService: ConsignmentService,
-    val frontEndInfoConfiguration: FrontEndInfoConfiguration
+    val frontEndInfoConfiguration: ApplicationConfig
```
Comment (vimleshtna) [inline app/controllers/FileChecksController.scala:22]: Can you please rename `frontEndInfoConfiguration` to `applicationConfig`?

**Review by vimleshtna — COMMENTED** [review]
Looks good, just a few minor comments, thanks.

File: app/controllers/FileChecksResultsController.scala (hunk 3 above)
Comment (MancunianSam) [reply app/controllers/FileChecksResultsController.scala:23]: Sorry, I was rushing. I've removed this and updated the tests.

**Review by vimleshtna — APPROVED** [review]
Looks good, thanks.

**Pull request merged**